### ISSUE TITLE
Mac M1 use Godot_v4.0-rc3_mono_macos open project encounter crash ！

### ISSUE THREAD

**Issue by ceragon** [godotengine/godot#73747]
### Godot version

4.0.dev-rc3

### System information

MacOS 12.6.3，M1 Chip

### Issue description

I try use Godot_v4.0-rc3_mono_macos to open a project that is generate by Godot V3.5.1，then i encounter crash。I reopen the project，but crash again。
![20230222172145](https://user-images.githubusercontent.com/6135880/220577080-7aba7e72-dd18-48de-adca-f8ee003463a5.jpg)

### Steps to reproduce

1. use Godot V3.5.1 create a project（Official 2D Demo）
2. use Godot_v4.0-rc3_mono_macos to open the project
3. then Godot crash

### Minimal reproduction project

[OfficialDemo.zip](https://github.com/godotengine/godot/files/10802022/OfficialDemo.zip)

**Comment by MikeEngler** [comment]
I had similar problem today and at least for me the solution was to install [.net7](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) as you can read [here](https://github.com/godotengine/godot/issues/73756)

**Comment by ceragon** [comment]
> I had similar problem today and at least for me the solution was to install [.net7](https://dotnet.microsoft.com/en-us/download/dotnet/7.0) as you can read [here](https://github.com/godotengine/godot/issues/73756). That should help with the crashing.

I upgrade to dotnet7，and update mono to latest version，but also crashing。: (
![20230223111935](https://user-images.githubusercontent.com/6135880/220814609-e0e87816-7de1-49b0-b6ab-08df02070018.jpg)

**Comment by akien-mga** [comment]
Can you post the output of `dotnet --info`?

**Comment by ceragon** [comment]
> Can you post the output of `dotnet --info`?

ceragon@ceragondeMacBook-Air ~ % dotnet --info
.NET SDK:
 Version:   7.0.100
 Commit:    e12b7af219

运行时环境:
 OS Name:     Mac OS X
 OS Version:  12.6
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /opt/homebrew/Cellar/dotnet/7.0.100/libexec/sdk/7.0.100/

Host:
  Version:      7.0.0
  Architecture: arm64
  Commit:       d099f075e4

.NET SDKs installed:
  7.0.100 [/opt/homebrew/Cellar/dotnet/7.0.100/libexec/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.0 [/opt/homebrew/Cellar/dotnet/7.0.100/libexec/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.0 [/opt/homebrew/Cellar/dotnet/7.0.100/libexec/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  DOTNET_ROOT       [/opt/homebrew/Cellar/dotnet/7.0.100/libexec]

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Comment by ceragon** [comment]
I solve the problem by download and install .net 6.0 from Microsoft official site。
The steps to identify the problem are as follows:

1. download [macos godot engine -.net 4.0](https://godotengine.org/download/macos/) from the official site。
2. open the godot app and get a prompt
![image](https://user-images.githubusercontent.com/6135880/222331763-8836cd84-9758-41f7-9b60-a20cf0ee9e28.png)
3. I actually have installed the .net 7.0 environment through homebrew and configured to PATH environment variables, but godot seem did not recognize env。
```
dotnet --info
.NET SDK:
Version: 7.0.100
Commit: e12b7af219

运行时环境:
OS Name: Mac OS X
OS Version: 12.6
OS Platform: Darwin
RID: osx.12-arm64
Base Path: /opt/homebrew/Cellar/dotnet/7.0.100/libexec/sdk/7.0.100/

```
4. In the end, I had to accept its suggestion to download and install the .net environment from Microsoft's official website
5. finally, I successfully opened the project through godot 4.0. 
```
/usr/local/share/dotnet/dotnet --info

.NET SDK:
 Version:   6.0.406
 Commit:    2988897946

运行时环境:
 OS Name:     Mac OS X
 OS Version:  12.6
 OS Platform: Darwin
 RID:         osx.12-arm64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.406/
```


Although I solved the problem, I think godot may ignore the .net installed through homebrew and the environment variables. Hope this opinion can help you! Thanks :)
